### DISCUSSION TITLE
path output into FromPath

### DISCUSSION THREAD

**Discussion by RamilNurtdinov** [nextflow-io/nextflow#3510] [General]
Hi

I use DSL2
nextflow.enable.dsl=2

I have a process that makes a file and return it as path variable
```
output:
        path "Expressed.genes.tsv"
```
I want to read this file with FromPath 
```
    channel
        .fromPath(file(genes_index))
        .splitCsv(header:true, sep: '\t')
```

I got error:
class groovyx.gpars.dataflow.expression.DataflowInvocationExpression cannot be cast to class java.nio.file.FileSystem (groovyx.gpars.dataflow.expression.DataflowInvocationExpression is in unnamed module of loader 'app'; java.nio.file.FileSystem is in module java.base of loader 'bootstrap'

Eventually I need to emit a channel with every line from the "path" object.

Be grateful if somebody can help me with it.

Ramil
.

**Comment by pditommaso** [comment]
I guess `genes_index` is already a channel therefore you don't need (and can't) to use `channel.fromPath`. Try instead 

```
genes_index.splitCsv(header:true, sep: '\t')
```

**Reply by RamilNurtdinov** [reply]
Thank you Paolo
This works.

I am running the following command:
`expressed_genes = Expressed_genes(data_folder,tissues,chunks,s_status) | splitCsv(header:true, sep: '\t') | toList()`
The `Expressed_genes` process returns `output: path "Expressed.genes.tsv"` and the content of this file is redirected to `splitCsv` somehow.

From my point of view you need to update the manual. 
Either the `path` variable can be simultaneously the content of its file, or `splitCsv` understands the path variable and reads corresponding file.
